File: CJUIKitDemo/CJUIKitDemo/Modules/DemosFoundation/Insert/TSInsertFeedHomeViewController.m
//
//  TSInsertFeedHomeViewController.m
//  CJFoundationDemo
//
//  Created by ciyouzen on 2016/3/26.
//  Copyright © 2016年 dvlproad. All rights reserved.
//

#import "TSInsertFeedHomeViewController.h"

#import "CJUIKitDemo-Swift.h"


@interface TSInsertFeedHomeViewController () {
    
}
@property (nonatomic, strong) TSFeedAdInsertUtil *feedAdInsertUtil1;
@property (nonatomic, strong) TSFeedAdInsertUtil *feedAdInsertUtil2;

@property (nonatomic, strong) TSFeedAdInsertUtil *feedAdInsertUtil11;
@property (nonatomic, strong) NSMutableArray<TSTempDataModel *> *currentDataModels;


@end

@implementation TSInsertFeedHomeViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.title = NSLocalizedString(@"广告流插入测试首页", nil);
    
    self.feedAdInsertUtil1 = [[TSFeedAdInsertUtil alloc] initWithCurrentDataModels:@[] afterEveryRowSteps:@[] remainderEveryRowStep:4 itemCountPerRow:1];
    self.feedAdInsertUtil2 = [[TSFeedAdInsertUtil alloc] initWithCurrentDataModels:@[] afterEveryRowSteps:@[@2] remainderEveryRowStep:1 itemCountPerRow:3];
    
    NSMutableArray *currentDataModels = [[NSMutableArray alloc] init];
    for (int i = 0; i < 50; i++) {
        TSTempDataModel *tempDataModel = [[TSTempDataModel alloc] init];
        [currentDataModels addObject:tempDataModel];
    }
    self.currentDataModels = currentDataModels;
    self.feedAdInsertUtil11 = [[TSFeedAdInsertUtil alloc] initWithCurrentDataModels:self.currentDataModels afterEveryRowSteps:@[@2] remainderEveryRowStep:1 itemCountPerRow:3];
    
    NSMutableArray *sectionDataModels = [[NSMutableArray alloc] init];
    
    // FeedAd
    {
        CQDMSectionDataModel *sectionDataModel = [[CQDMSectionDataModel alloc] init];
        sectionDataModel.theme = @"广告流FeedAd插入相关";
        {
            CQDMModuleModel *dateModule = [[CQDMModuleModel alloc] init];
            dateModule.title = @"广告流FeedAd插入相关--在分页数据中添加";
            dateModule.content = [self.feedAdInsertUtil1 getInsertDescription];
            dateModule.contentLines = 4;
            dateModule.actionBlock = ^{
                [self.feedAdInsertUtil1 insertElementsToAppendArray];
            };
            [sectionDataModel.values addObject:dateModule];
        }
        {
            CQDMModuleModel *dateModule = [[CQDMModuleModel alloc] init];
            dateModule.title = @"广告流FeedAd插入相关--在分页数据中添加";
            dateModule.content = [self.feedAdInsertUtil2 getInsertDescription];
            dateModule.contentLines = 4;
            dateModule.actionBlock = ^{
                [self.feedAdInsertUtil2 insertElementsToAppendArray];
            };
            [sectionDataModel.values addObject:dateModule];
        }
        {
            CQDMModuleModel *dateModule = [[CQDMModuleModel alloc] init];
            dateModule.title = @"广告流FeedAd插入相关--整理自身";
            dateModule.content = [self.feedAdInsertUtil11 getInsertDescription];
            dateModule.contentLines = 4;
            dateModule.actionBlock = ^{
                [self.feedAdInsertUtil11 insertElementsToSelf];
            };
            [sectionDataModel.values addObject:dateModule];
        }
        
        [sectionDataModels addObject:sectionDataModel];
    }
    
    
    self.sectionDataModels = sectionDataModels;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

/*
#pragma mark - Navigation

// In a storyboard-based application, you will often want to do a little preparation before navigation
- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    // Get the new view controller using [segue destinationViewController].
    // Pass the selected object to the new view controller.
}
*/

@end

File: TSDemo_DataVientiane-Swift/InsertDemo/TSInsertFeedHomeViewController.swift
//
//  TSInsertFeedHomeViewController.swift
//  CJUIKitDemo
//
//  Converted from Objective-C on 2026/7/16.
//

import UIKit
import CQDemoKit

@objc public class TSInsertFeedHomeViewController: CJUIKitBaseHomeViewController {

    private var feedAdInsertUtil1: TSFeedAdInsertUtil!
    private var feedAdInsertUtil2: TSFeedAdInsertUtil!
    private var feedAdInsertUtil11: TSFeedAdInsertUtil!
    private var currentDataModels: [TSTempDataModel] = []

    public override func viewDidLoad() {
        super.viewDidLoad()

        self.navigationItem.title = NSLocalizedString("广告流插入测试首页", comment: "")

        self.feedAdInsertUtil1 = TSFeedAdInsertUtil(
            currentDataModels: [],
            afterEveryRowSteps: [],
            remainderEveryRowStep: 4,
            itemCountPerRow: 1
        )
        self.feedAdInsertUtil2 = TSFeedAdInsertUtil(
            currentDataModels: [],
            afterEveryRowSteps: [2],
            remainderEveryRowStep: 1,
            itemCountPerRow: 3
        )

        var models: [TSTempDataModel] = []
        for _ in 0..<50 {
            models.append(TSTempDataModel())
        }
        self.currentDataModels = models
        self.feedAdInsertUtil11 = TSFeedAdInsertUtil(
            currentDataModels: self.currentDataModels,
            afterEveryRowSteps: [2],
            remainderEveryRowStep: 1,
            itemCountPerRow: 3
        )

        var sectionDataModels: [CQDMSectionDataModel] = []

        // FeedAd
        let sectionDataModel = CQDMSectionDataModel()
        sectionDataModel.theme = "广告流FeedAd插入相关"

        let module1 = CQDMModuleModel()
        module1.title = "广告流FeedAd插入相关--在分页数据中添加"
        module1.content = self.feedAdInsertUtil1.getInsertDescription()
        module1.contentLines = 4
        module1.actionBlock = { [weak self] in
            self?.feedAdInsertUtil1.insertElementsToAppendArray()
        }
        sectionDataModel.values.add(module1)

        let module2 = CQDMModuleModel()
        module2.title = "广告流FeedAd插入相关--在分页数据中添加"
        module2.content = self.feedAdInsertUtil2.getInsertDescription()
        module2.contentLines = 4
        module2.actionBlock = { [weak self] in
            self?.feedAdInsertUtil2.insertElementsToAppendArray()
        }
        sectionDataModel.values.add(module2)

        let module3 = CQDMModuleModel()
        module3.title = "广告流FeedAd插入相关--整理自身"
        module3.content = self.feedAdInsertUtil11.getInsertDescription()
        module3.contentLines = 4
        module3.actionBlock = { [weak self] in
            self?.feedAdInsertUtil11.insertElementsToSelf()
        }
        sectionDataModel.values.add(module3)

        sectionDataModels.append(sectionDataModel)

        self.sectionDataModels = NSMutableArray(array: sectionDataModels)
    }
}
